[Tsenrae] iP

Tsenrae opened the pull request:
### Bozo

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Bozo frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/joelgoh1/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😆

And it is **FREE!**

Features:


- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public static void main(String[] args) {
        new Bozo().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Tsenrae committed:
Level-0

Tsenrae committed:
Added echo functionality

Tsenrae committed:
added add and list functionality

Tsenrae committed:
added mark as done/undone functionality

Tsenrae committed:
Added ToDo, Event and Deadline

Tsenrae committed:
Added Automated Text UI Testing

Tsenrae committed:
Added Exception Handling for empty tasks and unrecognised commands

Tsenrae committed:
Added delete functionality

Tsenrae committed:
Added save functionality

Tsenrae committed:
Added usage of LocalDateTime to read dates/times for events and deadlines

Tsenrae committed:
Added Storage, Parser, Ui and TaskList classes

Tsenrae committed:
Organised all classes into one package bozo

Tsenrae committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

Tsenrae committed:
Added gradle support

Tsenrae committed:
Added JUnit tests to Todo and TaskList, named TodoTest and TaskListTest

Tsenrae committed:
Packaged bozo as Jar file

Tsenrae committed:
Added functionality to find a task by saying a keyword (eg. "find" would find a task like "read book"

Tsenrae committed:
Added functionality to find a task using keywords

Tsenrae committed:
Revert "Added functionality to find a task using keywords"

This reverts commit 5ea501dcfcf3a58ec01d927b32b08506266fa6ef.

Tsenrae committed:
Wrong branch name earlier for level-9, changed it to branch-Level-9

Tsenrae committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bozo/Parser.java

Tsenrae committed:
Add javaDocs to more than half of the non-private classes/methods

Classes include Bozo, BozoException, Deadline, Event, Parser, Storage,
TaskList, Todo

Joelwang22 commented on src/test/java/bozo/TaskListTest.java:
```diff
@@ -0,0 +1,31 @@
+
+public class TaskListTest {
+    @Test
+    public void testAddTask() {
```
I like how thorough you are with test cases! However I would like to know whether the tests are testing for success or failure, this can be shown by altering the name to be like Todo_addTask_success() or Todo_removeTask_success() so we know that we are testing for success cases and not fail cases.

Joelwang22 reviewed (commented):
Looks great!
some naming changes can make it look better 👍

Joelwang22 commented on src/main/java/bozo/Ui.java:
```diff
@@ -0,0 +1,45 @@
+public class Ui {
+
+    public void showWelcome() {
+        String logo = ".-. .-')                  .-') _ \n"
+                + "\\  ( OO )                (  OO) ) \n"
+                + " ;-----.\\  .-'),-----. ,(_)----. .-'),-----. \\\n"
+                + " | .-.  | ( OO'  .-.  '|       |( OO'  .-.  '\n"
+                + " | '-' /_)/   |  | |  |'--.   / /   |  | |  |\n"
+                + " | .-. `. \\_) |  |\\|  |(_/   /  \\_) |  |\\|  |\n"
+                + " | |  \\  |  \\ |  | |  | /   /___  \\ |  | |  |\n"
+                + " | '--'  /   `'  '-'  '|        |  `'  '-'  '\n"
+                + " `------'      `-----' `--------'    `-----'\n";
+
+        String logo2 = "▄▄▄▄·       ·▄▄▄▄•\n"
+                + " ▐█ ▀█▪▪     ▪▀·.█▌▪\n"
+                + " ▐█▀▀█▄ ▄█▀▄ ▄█▀▀▀• ▄█▀▄\\\n"
+                + " ██▄▪▐█▐█▌.▐▌█▌▪▄█▀▐█▌.▐▌\n"
+                + ".▀▀▀▀  ▀█▄▀▪·▀▀▀ • ▀█▄▀▪\n";
```
Nice Logos! I like the creativity 💯

Joelwang22 reviewed (commented):
Looks Good, I like what I see 🥇

Tsenrae committed:
Tweak the code to adhere to the coding standard

Changes include shortening variable names in Storage Class and
correcting indentation for switch statements

Tsenrae committed:
Add checkstyle functionality

matthewken19 commented on src/main/java/bozo/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
+    protected static DateTimeFormatter input = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    protected static DateTimeFormatter output = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
```
I think static constants should be in uppercase (i.e. INPUT, OUTPUT)

matthewken19 commented on src/main/java/bozo/Storage.java:
```diff
@@ -0,0 +1,93 @@
+                String s = parts[2];
+
+                switch (type) {
+                case "T":
+                    Todo td = new Todo(s);
+                    if (i == 1) {
+                        td.isDone = true;
+                    } else {
+                        td.isDone = false;
+                    }
+                    list.add(td);
+                    break;
+                case "D":
+                    Deadline d = new Deadline(s, parts[3]);
+                    if (i == 1) {
+                        d.isDone = true;
+                    } else {
+                        d.isDone = false;
+                    }
+                    list.add(d);
+                    break;
+                case "E":
+                    Event e = new Event(s, parts[3], parts[4]);
+                    if (i == 1) {
+                        e.isDone = true;
+                    } else {
+                        e.isDone = false;
+                    }
+                    list.add(e);
+                    break;
+                }
```
i think the if else statement can be moved down to after the switch statement and assign the todo, deadline, event object to Task, then do one if else on that task.

matthewken19 reviewed (commented):
Overall, I like the OOP abstraction! LGTM, but just take note of some changes that I have proposed that might help!

meowwtama commented on src/main/java/bozo/Bozo.java:
```diff
@@ -0,0 +1,50 @@
+    private static Parser parser;
+
+    /**
+     * Constructor for the Bozo class.
```
Might want to add a line space between the description and the params.

meowwtama commented on src/main/java/bozo/Task.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Represents a task.
+ */
+public class Task {
```
Can consider using an abstract class.

meowwtama commented on src/main/java/bozo/TaskList.java:
```diff
@@ -0,0 +1,72 @@
+ */
+public class TaskList implements Iterable<Task> {
+
+    public ArrayList<Task> list;
```
The variable name can be more descriptive.

meowwtama commented on src/main/java/bozo/Ui.java:
```diff
@@ -0,0 +1,45 @@
+                + " | .-. `. \\_) |  |\\|  |(_/   /  \\_) |  |\\|  |\n"
+                + " | |  \\  |  \\ |  | |  | /   /___  \\ |  | |  |\n"
+                + " | '--'  /   `'  '-'  '|        |  `'  '-'  '\n"
+                + " `------'      `-----' `--------'    `-----'\n";
```
I really like this logo! It captivates with its simplicity and depth, a perfect blend of modern design and timeless elegance.

meowwtama commented on src/main/java/bozo/Bozo.java:
```diff
@@ -0,0 +1,50 @@
+ * It contains the main method and the run method.
+ */
+public class Bozo {
+    private static TaskList list;
```
Can consider a more descriptive name for the tasklist!

meowwtama reviewed (commented):
LGTM mostly well done! The logo looks really creative!! Just some small issues I noted.

Tsenrae committed:
Add GUI to bozo chatbot using javaFX

Tsenrae committed:
Add FXML to GUI

Add FXML files and dependencies

Tsenrae committed:
Create new JAR file

Jar file now contains GUI for Bozo and additional updates

These include find() method, javaDocs, checkStyle functionality

Tsenrae committed:
Add assertions to Storage and Parser classes

For storage class, assertions include checking whether list is empty
before saving or loading tasks, whether file is readable and writable,
and whether the filePath exists

For Parser class, assertions include checking whether list is empty
before performing any actions (list, delete, find...) and whether the
user input is empty to ensure correct parsing of commands

Tsenrae committed:
Improve coding quality for Storage and Parser classes

For Storage class, perform abstraction on parsing the task in loadFile()
so that method is overall shorter

For Parser class, perform abstraction on all the different commands
(eg. find, delete, mark) so that parseCommand() method is overall
shorter

Tsenrae committed:
Merge pull request #2 from Tsenrae/branch-A-Assertions

Add assertions to Storage and Parser classes

Tsenrae committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/bozo/Parser.java

Tsenrae committed:
Merge pull request #3 from Tsenrae/branch-A-CodeQuality

Improve coding quality for Storage and Parser classes

Tsenrae committed:
Add friendlier syntax for adding tasks

Changes include "t" for "todo", "d" for "deadline" and "e" for event

Emberlynn-Loo commented on src/main/resources/images/DaBozo.png:
I love that you incorporated different pictures from the tutorial especially since it matches your chatbot name.

Emberlynn-Loo commented on src/main/java/bozo/Parser.java:
```diff
@@ -0,0 +1,230 @@
+
+        if (input.equals("bye")) {
+            return handleByeCommand();
+        } else if (input.equals("list")) {
```
Any reason you chose to use else of instead of switch-cases?

Emberlynn-Loo reviewed (commented):
Great code overall!

Tsenrae committed:
Add enhancements to GUI and screenshot of product

Tsenrae committed:
Add chat messaging bubble for dialogBoxes and update user guide

Tsenrae committed:
Update user guide

Tsenrae committed:
Merge pull request #4 from Tsenrae/branch-A-UserGuide

Add chat messaging bubble for dialogBoxes and update user guide

Tsenrae committed:
Merge branch 'master' into branch-UserGuide

Tsenrae committed:
Merge pull request #5 from Tsenrae/branch-UserGuide

Branch user guide

Tsenrae committed:
Update build.gradle file to enable JAR file to run smoothly on all
platforms
